keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,485 @@
+
+## Motivation
+
+For end-users who would like to run unprivileged containers using user namespaces _nested inside_ CRI containers, we need an option to have a `ProcMount`.
+That is, we need an option to designate explicitly turn off masking and setting read-only of paths so that we can mount `/proc` in the nested container as an unprivileged user.
+
+Please see the following filed issues for more information:
+- [opencontainers/runc#1658](https://github.com/opencontainers/runc/issues/1658#issuecomment-373122073)
+- [moby/moby#36597](https://github.com/moby/moby/issues/36597)
+- [moby/moby#36644](https://github.com/moby/moby/pull/36644)
+
+### Goals
+
+- Add the ProcMountType option to the pod API, allowing users to override the proc mount behavior in a container.
+
+### Non-Goals
+
+## Proposal
+
+
+Add a new `string` type field named `ProcMountType` will hold the viable
+options for `procMount` to the `SecurityContext`
+definition.
+
+By default,`procMount` is `default`, aka the same behavior as today and the
+paths are masked.
+
+This will look like the following in the spec:
+
+```go
+type ProcMountType string
+
+const (
+    // DefaultProcMount uses the container runtime default ProcType.  Most 
+    // container runtimes mask certain paths in /proc to avoid accidental security
+    // exposure of special devices or information.
+    DefaultProcMount ProcMountType = "Default"
+
+    // UnmaskedProcMount bypasses the default masking behavior of the container
+    // runtime and ensures the newly created /proc the container stays in tact with
+    // no modifications.  
+    UnmaskedProcMount ProcMountType = "Unmasked"
+)
+
+procMount *ProcMountType
+```
+
+where nil is default, and is interpreted as "Default" ProcMountType.
+
+When the kubelet is presented with a pod that has a ProcMountType as Unmasked, it will edit the default list of
+masked paths it passes down to the [CRI to be empty](https://github.com/kubernetes/kubernetes/blob/964529b/pkg/securitycontext/util.go#L216) which it does
+with the [CRI request](https://github.com/kubernetes/kubernetes/blob/964529b/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto#L889-L891).
+
+This requires changes to the CRI runtime integrations so that kubelet will add the specific `unmasked` option.
+This was done after alpha:
+CRI-O has support in v1.25.0 after https://github.com/cri-o/cri-o/pull/6025/commits/4102586132214263c5d0ae93ec257432653ab82b
+containerd has support in 1.6. See https://github.com/containerd/containerd/pull/5070/commits/07f1df4541d6a81c205d194f4f6ea3a6a95c3e29
+
+The main use case for unmasking paths in `/proc` are for a user nesting unprivileged containers within a container. However, having an Unmasked ProcMountType
+is a privileged operation, and thus is part of the [privileged PSS policy](https://github.com/kubernetes/kubernetes/pull/103340). Since a user must have be in
```

Comment:
```suggestion
is a privileged operation, and thus is part of the [privileged](https://k8s.io/docs/concepts/security/pod-security-standards/#privileged) Pod Security Standard. Since a user must have be in
```
PodSecurity<i>Policies</i> are no longer part of Kubernetes

Reply:
(addressed)